src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/PackageRestoreDataSource.cs

```diff
@@ -85,6 +91,8 @@ internal class PackageRestoreDataSource : ChainedProjectValueDataSourceBase<Rest
             _fileSystem = fileSystem;
             _logger = logger;
             _projectDependentFileChangeNotificationService = projectDependentFileChangeNotificationService;
+
+            Name = _project.FullPath;
```

Comment:
don't cache the name on your side, but wire it directly. It reduces problems if FullPath changes (like renaming), even if it is not a scenario today.